### DISCUSSION TITLE
EPUBCheck v4.2.5 wrong path for java?

### DISCUSSION THREAD

**Discussion by shoesforindustry** [Homebrew/discussions#1204] [Everyday usage] [answered]
On my MAC (Big Sur) 11.2.3 (20D91) and EPUBCheck v4.2.5 from Homebrew
My recent upgrade of epubcheck to v4.2.5 started failing with:

`/usr/local/bin/epubcheck: line 3: /bin/java: No such file or directory /usr/local/bin/epubcheck: line 3: exec: /bin/java: cannot execute: No such file or directory`

`usr/local/bin/epubcheck` seems to have a path for the exec of `/bin/java` however Java on my Mac (installed from https://java.com/en/download/) is located in `/usr/bin/java`. I cannot seem to symlink it to /bin/java.

If I change the epubcheck file with the correct java path it works.

`#!/bin/bash export JAVA_HOME=“${JAVA_HOME:-}” exec “${JAVA_HOME}/usr/bin/java” -jar “/usr/local/Cellar/epubcheck/4.2.5/libexec/epubcheck.jar” “$@“`

Anyone any ideas why this is just suddenly happening?

**Comment by SMillerDev** [comment, marked as answer]
You should set `JAVA_HOME`, it's a standard variable to indicate the location of the Java JDK/JRE and clearly the latest EPUBCheck (updated 8 days ago) can't work without it.

**Reply by shoesforindustry** [reply]
Thanks SMIllerDev, adding 
`export JAVA_HOME="$(/usr/libexec/java_home)"`
to my bash_profile worked, much appreciate your help.
